client/asset/btc/btc.go

```diff
@@ -5089,3 +5091,71 @@ func deserializeMsgTx(r io.Reader) (*wire.MsgTx, error) {
 func msgTxFromBytes(txB []byte) (*wire.MsgTx, error) {
 	return deserializeMsgTx(bytes.NewReader(txB))
 }
+
+func (btc *baseWallet) ConfirmRedemption(coinID dex.Bytes, redemption *asset.Redemption) (*asset.ConfirmRedemptionStatus, error) {
+	txHash, _, err := decodeCoinID(coinID)
+	if err != nil {
+		return nil, err
+	}
+
+	_, confs, err := btc.rawWalletTx(txHash)
+	// redemption transaction found, return its confirms.
+	if err == nil {
+		return &asset.ConfirmRedemptionStatus{
+			Confs:  uint64(confs),
+			Req:    reqiredRedeemConfirms,
+			CoinID: coinID,
+		}, nil
+	}
+
+	if !errors.Is(err, WalletTransactionNotFound) {
+		return nil, fmt.Errorf("problem searching for redemption transaction %s: %v", txHash, err)
+	}
+
+	// Redemption transaction is missing from the point of view of our node!
+	// Unlikely, but possible it was redeemed by another transaction. We
+	// assume a contract past its locktime cannot make it here, so it must
+	// not be refunded. Check if the contract is still an unspent output.
+
+	pkScript, err := btc.scriptHashScript(redemption.Spends.Contract)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing pubkey script: %v", err)
+	}
+
+	txHash, vout, err := decodeCoinID(redemption.Spends.Coin.ID())
+	if err != nil {
+		return nil, err
+	}
+
+	utxo, _, err := btc.node.getTxOut(txHash, vout, pkScript, time.Now().Add(-ContractSearchLimit))
+	if err != nil {
+		return nil, fmt.Errorf("error finding unspent contract %s: %w", redemption.Spends.Coin.ID(), err)
+	}
+	if utxo == nil {
+		// TODO: Spent, but by who. Find the spending tx.
```

Comment:
I actually ran into a similar situation with someone in production where their client has actually redeemed, but because of a dcrd timeout with dcrwallet, Core's call to Redeem returned an error, and Core didn't record the redeem that really did broadcast.  As a result it retried a redeem repeatedly only to fail because the contract output was already spent.

However in the case I'm addressing we wouldn't be trying to confirm a known redemption yet, so it would have to be caught in one of the other ticking checks (isRedeemable).  I recently made a comment in the bonds pr diff where we'd do this:  https://github.com/decred/dcrdex/pull/2036/commits/8e0761761256a94ca17f2a1b59450bd05aab1086#diff-8962b4c526755f0f98ed9f46ae05827e8edbc43ed5671ea4a2624bdf1617496cR1565

Anyway, finding our own (forgotten) redeem would of course be similar to the recent refund search like in https://github.com/decred/dcrdex/pull/2011/files
I had tentatively planned to do similar to that if no one else go to that first.  I suppose the same code would be used for this TODO in `ConfirmRedemption`